### PR TITLE
feat(Table): Add `rowsPerPageLabel` to paginator localization

### PR DESCRIPTION
**Pull request by mayank99**
Added `rowsPerPageLabel` property to paginator localization to show label for the page size selector. Used hook from #436 to hide label on small widths. User can also pass empty string if they don't want the label.

## Checklist

- [x] Add meaningful unit tests for your component (verify that all lines are covered)
- [x] Verify that all existing tests pass
- [ ] Add component features demo in Storybook (different stories)
- [ ] Approve test images for new stories
- [ ] Add screenshots of the key elements of the component

### CODE REVIEW

File: src/core/Table/TablePaginator.tsx [hunk 1]
```diff
@@ -311,6 +331,11 @@ export const TablePaginator = (props: TablePaginatorProps) => {
         </IconButton>
       </div>
       <div className='iui-right'>
+        {localization.rowsPerPageLabel && currentWidth >= 1024 && (
```
Comment (mayank99) [inline src/core/Table/TablePaginator.tsx:334]: I am not sure if this should be 1024. Might be a bit too much. But 768 might be too little, idk

File: src/core/Table/TablePaginator.test.tsx [hunk 2]
```diff
@@ -296,12 +297,21 @@ it('should render with custom localization', () => {
...
   const pageSizeSelector = container.querySelector(
     '.iui-dropdown',
   ) as HTMLButtonElement;
   expect(pageSizeSelector).toBeTruthy();
   expect(pageSizeSelector.textContent).toEqual('1-10 of test 195');
 
+  // TODO: fix this
+  // expect(
+  //   container.querySelector('.iui-right .iui-text-block'),
+  // ).toHaveTextContent('Items per test page');
```
Comment (mayank99) [inline src/core/Table/TablePaginator.test.tsx:313]: I don't know an easy way to test this

File: src/core/Table/TablePaginator.tsx [hunk 3]
```diff
@@ -177,6 +186,13 @@ export const TablePaginator = (props: TablePaginatorProps) => {
   );
   const [overflowRef, visibleCount] = useOverflow(pageList);
 
+  const paginatorRef = React.useRef<HTMLDivElement>(null);
+  const [currentWidth, setCurrentWidth] = React.useState(
+    () => paginatorRef.current?.offsetWidth ?? 0,
+  );
+  const [resizeRef] = useResizeObserver(({ width }) => setCurrentWidth(width));
+  const paginatorRefs = useMergedRefs(paginatorRef, resizeRef);
```
Comment (mayank99) [inline src/core/Table/TablePaginator.tsx:194]: This is actually buggy, which is why the test is failing. Once #436 is merged, I can just reuse the logic from there

File: src/core/Table/TablePaginator.tsx [hunk 4]
```diff
@@ -311,6 +325,11 @@ export const TablePaginator = (props: TablePaginatorProps) => {
         </IconButton>
       </div>
       <div className='iui-right'>
+        {localization.rowsPerPageLabel && paginatorWidth >= 1024 && (
+          <Text isMuted as='span'>
+            {localization.rowsPerPageLabel}
```
Comment (saaaaaally) [inline src/core/Table/TablePaginator.tsx:330]: May need to do something to handle very long strings
image
image

File: src/core/Table/TablePaginator.tsx [hunk 5]
```diff
@@ -311,6 +325,11 @@ export const TablePaginator = (props: TablePaginatorProps) => {
         </IconButton>
       </div>
       <div className='iui-right'>
+        {localization.rowsPerPageLabel && paginatorWidth >= 1024 && (
```
Comment (saaaaaally) [inline src/core/Table/TablePaginator.tsx:328]: Label shouldn't be shown if pageSizeList is undefined

File: src/core/Table/TablePaginator.tsx [hunk 6]
```diff
@@ -89,6 +92,11 @@ export type TablePaginatorProps = {
      * @default (page: number) => `Go to page ${page}`
      */
     goToPageLabel?: (page: number) => string;
+    /**
+     * A label shown next to the page size selector. Use `null` to hide.
+     * @default 'Rows per page'
+     */
+    rowsPerPageLabel?: string | null;
```
Comment (saaaaaally) [inline src/core/Table/TablePaginator.tsx:99]: How come this label doesn't change the visual test images?

File: src/core/Table/TablePaginator.tsx (hunk 5 above)
Comment (mayank99) [reply src/core/Table/TablePaginator.tsx]: Fixed by merging main

File: src/core/Table/TablePaginator.tsx (hunk 6 above)
Comment (mayank99) [reply src/core/Table/TablePaginator.tsx:99]: It's because the table is not wide enough in the test environment.

File: src/core/Table/TablePaginator.tsx (hunk 4 above)
Comment (mayank99) [reply src/core/Table/TablePaginator.tsx]: this is easily fixed with `text-overflow: ellipsis` but that will need to be in CSS

Comment (mayank99) [reply src/core/Table/TablePaginator.tsx]: Added `.iui-paginator-page-size-label` from https://github.com/iTwin/iTwinUI/pull/381. Doesn't do anything currently as it needs CSS update (Not updating CSS in this PR because it will break buttons)

File: src/core/Table/TablePaginator.tsx [hunk 7]
```diff
@@ -89,6 +92,11 @@ export type TablePaginatorProps = {
      * @default (page: number) => `Go to page ${page}`
      */
     goToPageLabel?: (page: number) => string;
+    /**
+     * A label shown next to the page size selector. Use `null` to hide.
```
Comment (saaaaaally) [inline src/core/Table/TablePaginator.tsx:96]: Maybe add test that label is hidden when passing in null or empty string

File: src/core/Table/TablePaginator.tsx [hunk 8]
```diff
@@ -320,31 +334,38 @@ export const TablePaginator = (props: TablePaginatorProps) => {
...
-            }
-          >
-            {localization.rangeLabel(
-              currentPage * pageSize + 1,
-              Math.min(totalRowsCount, (currentPage + 1) * pageSize),
-              totalRowsCount,
-              isLoading,
+          <>
+            {localization.rowsPerPageLabel && paginatorWidth >= 1024 && (
+              <Text as='span' className='iui-paginator-page-size-label'>
```
Comment (saaaaaally) [inline src/core/Table/TablePaginator.tsx:339]: Why not just use span here?

Comment (mayank99) [reply src/core/Table/TablePaginator.tsx]: changed to span

File: src/core/Table/TablePaginator.tsx (hunk 7 above)
Comment (mayank99) [reply src/core/Table/TablePaginator.tsx:95]: added test 🤷‍♀️

**Review by saaaaaally — APPROVED** [review]

File: src/core/Table/TablePaginator.tsx [hunk 9]
```diff
@@ -28,6 +34,7 @@ const defaultLocalization = {
   previousPage: 'Previous page',
   nextPage: 'Next page',
   goToPageLabel: (page: number) => `Go to page ${page}`,
+  rowsPerPageLabel: 'Rows per page',
```
Comment (bentleyvk) [inline src/core/Table/TablePaginator.tsx:37]: Why stories screenshot did not changed?

File: src/core/Table/TablePaginator.tsx [hunk 10]
```diff
@@ -89,6 +91,11 @@ export type TablePaginatorProps = {
      * @default (page: number) => `Go to page ${page}`
      */
     goToPageLabel?: (page: number) => string;
+    /**
```
Comment (bentleyvk) [inline src/core/Table/TablePaginator.tsx:94]: Story would be nice because people will miss whether it is possible to hide this label. Also, is it possible to increase story screenshot width that label would be visible there?
